tests/unit/test_credentials.py

```diff
@@ -587,10 +588,38 @@ def test_create_credential_resolver(self):
             'metadata_service_num_attempts': 'e',
             'profile': 'profilename',
         }
-        fake_session.get_config_variable = lambda x: config[x]
-        resolver = credentials.create_credential_resolver(fake_session)
+        self.session.get_config_variable = lambda x: self.config[x]
+
+    def test_create_credential_resolver(self):
+        resolver = credentials.create_credential_resolver(self.session)
         self.assertIsInstance(resolver, credentials.CredentialResolver)
 
+    def test_explicit_profile_ignores_env_provider(self):
+        self.config['profile'] = 'dev'
+        resolver = credentials.create_credential_resolver(self.session)
+
+        for provider in resolver.providers:
+            self.assertNotIsInstance(provider, credentials.EnvProvider)
+
+    def test_no_profile_checks_env_provider(self):
+        self.config['profile'] = None
+        self.session.profile = None
+        resolver = credentials.create_credential_resolver(self.session)
+
+        found = False
```

Comment:
Or to assert that all the providers are not an instance of the env provider, you can just use: `self.assertTrue(all(not isinstance(p, EnvProvider) for p in p.providers))`